src/ukify/ukify.py

```diff
@@ -1297,7 +1297,18 @@ def make_uki(opts: UkifyConfig) -> None:
     linux = opts.linux
     combined_sigs = '{}'
 
-    if opts.linux and sign_args_present:
+    # On some distros, on some architectures, the vmlinuz is a gzip file, so we need to decompress it
+    # if it's not a valid PE file, as it will fail to be booted by the firmware.
+    if linux:
+        try:
+            pefile.PE(linux, fast_load=True)
+        except pefile.PEFormatError:
+            decompressed = maybe_decompress(linux)
+            if decompressed:
+                linux = Path(tempfile.NamedTemporaryFile(prefix='linux-decompressed').name)
+                linux.write_bytes(decompressed)
```

Comment:
Might as well check if it's a PE file again now?

Reply:
aren't elf files also supported?